contracts/base/Dispatcher.sol

```diff
@@ -164,6 +163,23 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks
         }
     }
 
+    function approveAndSell721(bytes memory inputs, address protocol)
+        internal
+        returns (bool success, bytes memory output)
+    {
+        (bytes memory data, address token, address spender, uint256 id) =
+            abi.decode(inputs, (bytes, address, address, uint256));
+
+        // this approval is auto-wiped when the token is transferred
+        ERC721(token).approve(spender, id);
+
+        (success, output) = protocol.call(data);
+
+        // if !success manually revoke the approval
+        // what to do with the NFT??
```

Comment:
interested in thoughts on this... if the NFT is pulled in from Permit2, and then a sale fails... what do we do with the NFT

Reply:
what's the status of the NFT approvals before and after line 174? I guess it's a little tough to inspect without Permit2, but I would think it should return to its original status, which would be an approval to Permit2?